### ISSUE TITLE
comment-on-pr-number is ignored on push events

### ISSUE THREAD

**Issue by ulrich-giraud** [pulumi/actions#1177]
### What happened?

We upgraded our action version from 4.1.0 to 5.2.3 and noticed that we weren't getting any comment on PR when we merged.
Everything works correctly on pull_requests events but not anymore on push events and comment-on-number is set

### Example

```yaml
      - name: Retrieve current PR
        uses: jwalton/gh-find-current-pr@v1
        id: find_pr
        with:
          state: all

      - name: Run pulumi update
        uses: pulumi/actions@v5.2.3
        with:
          command: up
          stack-name: ${{ inputs.stack-name }}
          work-dir: ${{ inputs.work-dir }}
          cloud-url: '<redacted>'
          comment-on-pr-number: ${{ steps.find_pr.outputs.pr }}
          diff: true
          edit-pr-comment: false
```

### Output of `pulumi about`

CLI          
Version      3.116.0
Go Version   go1.22.3
Go Compiler  gc

Host     
OS       arch
Version  "rolling"
Arch     x86_64

Backend        
Name           ufmwk
URL            <redacted>
User           ulrich
Organizations  
Token type     personal

Pulumi locates its logs in /tmp by default

### Additional context

It sounds like the regression was introduced by this commit : https://github.com/pulumi/actions/commit/ca33bed63d58e10be6523f0184b5d368021a7d3d#diff-4fab5baaca5c14d2de62d8d2fceef376ddddcc8e9509d86cfa5643f51b89ce3dL114-R124

In fact, the code right now doesn't allow to comment on anything other than a pull_request event :

```typescript
  if (config.commentOnPrNumber || config.commentOnPr) {
    const isPullRequest = context.payload.pull_request !== undefined;
    if (isPullRequest) {  // handlePullRequestMessage isn't called if context.payload.pull_request is not set
      core.debug(`Commenting on pull request`);
      invariant(config.githubToken, 'github-token is missing.');
      handlePullRequestMessage(config, projectName, output);
    }
  }
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by tgummerer** [comment]
Thanks for opening this issue!  This does indeed look like a bug to me, and looks like an accidental regression in #813.  I opened a PR to try and fix it in https://github.com/pulumi/actions/pull/1178.

**Comment by tgummerer** [comment]
I just released v5.2.4 of the GitHub action, which should make this work again.  https://github.com/pulumi/actions/releases/tag/v5.2.4
